### PR TITLE
No special treatment for rdf:type

### LINKED ISSUE #3: Type triple is always created
The _Jena Model Step_ always creates a type triple  for the constructed resource.

When the _Resource rdf:type_ input field is empty, the subject of the triple is the base URI:
```ttl
ex:resource1 a <> .
```

This is because `JenaModelStep.createModel` [always adds it](https://github.com/nationalarchives/kettle-jena-plugins/blob/b34fa0d98698321df1ef3111e5478ca5e3b89250/src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java#L168):

https://github.com/nationalarchives/kettle-jena-plugins/blob/b34fa0d98698321df1ef3111e5478ca5e3b89250/src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java#L160-L168

I think the assumption that a resource is necessarily typed doesn't always hold.
The related assumption that a resource has a single type doesn't either.

### LINKED ISSUE #4: Resource rdf:type is redundant
There is a specific input field _Resource rdf:type_ on the _Jena Model Step_ UI.

This is redundant since the _Fields to RDF Properties_ list works for `rdf:type` as much as other predicates.

The list also doesn't assume (#3) that type is required or singular as does the type input.

### LINKED ISSUE #5: Can't create rdf:type from field
The _Resource rdf:type_ input field on the _Jena Model Step_ UI doesn't allow creating a type triple based on an input field value.
It only allows constant values and variable references.

The _Fields to RDF Properties_ list doesn't suffer from this shortcoming (#4).

### PR DESCRIPTION
**Pull request by langsamu**
Resolves #3, resolves #4 and resolves #5.

Removes the _Resource rdf:type_ input from the _Create Jena Model_ step UI, as well as code that makes it work.
Users can still create type statements using the _Fields to RDF Properties_ list on the same dialog.

### CODE REVIEW

File: src/test/java/uk/gov/nationalarchives/pdi/step/jena/model/EndToEndIT.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        final String pluginFolderPath = "target\\kettle-jena-plugins-2.1.0-SNAPSHOT-kettle-plugin";
+        StepPluginType.getInstance().getPluginFolders().add(new PluginFolder(pluginFolderPath, false, true));
+
+        KettleEnvironment.init();
+    }
+
+    @Test
+    public void can_create_rdf_type_statement() throws KettleException {
+        // TODO(SL): This should be a resource.
+        final String transformationFilePath = "C:\\Users\\samu\\Source\\repos\\kettle-jena-plugins\\src\\test\\resources\\can_create_rdf_type_statement.ktr";
```
Comment (adamretter) [inline src/test/java/uk/gov/nationalarchives/pdi/step/jena/model/EndToEndIT.java]: If you place this file in `src/test/resources/uk/gov/nationalarchives/pdi/step/jena/model`, you can then do something like:

```java
final URL resUrl = getClass().getResource("an_create_rdf_type_statement.ktr");
final Path resPath = Paths.get(resURL.toURI());
```

Comment (langsamu) [reply src/test/java/uk/gov/nationalarchives/pdi/step/jena/model/EndToEndIT.java]: Exactly wht I meant, thanks.

**Comment by langsamu** [comment]
Build fails due to hardcoded paths.
Fix pending #13.

File: src/test/java/uk/gov/nationalarchives/pdi/step/jena/model/EndToEndIT.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        KettleEnvironment.init();
+    }
+
+    @Test
+    public void can_create_rdf_type_statement() throws KettleException, IOException {
+        executeTransformation();
+        final Model actual = loadOutputGraph();
+        final Model expected = createExpectedModel();
+
+        assertTrue(actual.isIsomorphicWith(expected));
```
Comment (adamretter) [inline src/test/java/uk/gov/nationalarchives/pdi/step/jena/model/EndToEndIT.java:65]: Very nice!

**Comment by adamretter** [comment]
Perfect. Thank you @langsamu

**Pull request merged**